restli-common/src/main/java/com/linkedin/restli/common/util/ProjectionMaskApplier.java

```diff
@@ -215,6 +215,11 @@ else if (wildcardMask != null)
 
     UnionDataSchema newUnionDataSchema = new UnionDataSchema();
     newUnionDataSchema.setMembers(newUnionMembers, errorMessageBuilder);
+    if (newUnionMembers.size() > 0)
```

Comment:
This should be 
newMembers.size() < members.size() ?

Reply:
The `newMembers.size() < members.size()` could be including the case where there is no projection in the union. So I am hesitate to adopt this comment, let me know, Thanks

Reply:
Saw your latest comments